backend/routes/post.py

```
@@ -323,25 +308,29 @@ async def create_event_post(
+            # Generate a secure filename
+            secure_filename = f"{current_user.id}_{secrets.token_hex(8)}{file_extension}"
+            file_location = os.path.abspath(os.path.join(EVENT_UPLOAD_DIR, secure_filename))
+
+            # Prevent directory traversal
+            if not file_location.startswith(os.path.abspath(EVENT_UPLOAD_DIR)):
+                raise HTTPException(status_code=400, detail="Invalid file path detected.")
 
             # Save the uploaded image to the server
-            with open(file_path, "wb") as buffer:
-                shutil.copyfileobj(event_image.file, buffer)
+            with open(file_location, "wb") as buffer:
```

Ruff (0.8.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)